lib/unicore/mktables

```diff
@@ -7505,9 +7505,25 @@ END
                 push @{$loose_names_ending_in_code_point{$squeezed}->{'high'}},
                                                                          $high;
 
+                # Calculate the set of legal characters in names of this
+                # series.  It includes every character in the name prefix.
+                my %legal;
+                for my $i (0 .. length($map) - 1) {
+                    $legal{substr($map, $i, 1)} = 1;
+                }
```

Comment:
I'd have written that as `$legal{$_} = 1 for split //, $map;`.

Reply:
will change